code/game/gamemodes/clock_cult/clock_structures.dm

```diff
@@ -0,0 +1,768 @@
+		clockwork_component_cache["vanguard_cogwheel"] += S.stored_components["vanguard_cogwheel"]
+		clockwork_component_cache["guvax_capacitor"] += S.stored_components["guvax_capacitor"]
+		clockwork_component_cache["replicant_alloy"] += S.stored_components["replicant_alloy"]
+		clockwork_component_cache["hierophant_ansible"] += S.stored_components["hierophant_ansible"]
+		S.stored_components["belligerent_eye"] = 0
+		S.stored_components["vanguard_cogwheel"] = 0
+		S.stored_components["guvax_capacitor"] = 0
+		S.stored_components["replicant_alloy"] = 0
+		S.stored_components["hierophant_ansible"] = 0
+		user.visible_message("<span class='notice'>[user] empties [S] into [src].</span>", "<span class='notice'>You offload your slab's components into [src].</span>")
+		return 1
+	else if(istype(I, /obj/item/clockwork/daemon_shell))
+		var/component_type
+		switch(alert(user, "Will this daemon produce a specific type of component or produce randomly?.", , "Specific Type", "Random Component"))
+			if("Specific Type")
+				switch(input(user, "Choose a component type.", name) as null|anything in list("Belligerent Eyes", "Vanguard Cogwheels", "Guvax Capacitors", "Replicant Alloys", "Hierophant Ansibles"))
+					if("Belligerent Eyes")
+						component_type = "belligerent_eye"
+					if("Vanguard Cogwheels")
+						component_type = "vanguard_cogwheel"
+					if("Guvax Capacitors")
+						component_type = "guvax_capacitor"
+					if("Replicant Alloys")
+						component_type = "replicant_alloy"
+					if("Hierophant Ansibles")
+						component_type = "hierophant_ansibles"
+		if(!user || !user.canUseTopic(src) || !user.canUseTopic(I))
+			return 0
+		var/obj/item/clockwork/tinkerers_daemon/D = new(src)
+		D.cache = src
+		D.specific_component = component_type
+		user.visible_message("<span class='notice'>[user] spins the cogwheel on [I] and puts it into [src].</span>", \
+		"<span class='notice'>You activate the daemon and put it into [src]. It will now produce a component every thirty seconds.</span>")
+		user.drop_item()
+		qdel(I)
+		return 1
+	else
+		return ..()
+
+/obj/structure/clockwork/cache/attack_hand(mob/user)
+	if(!is_servant_of_ratvar(user))
+		return 0
+	var/list/possible_components = list()
+	if(clockwork_component_cache["belligerent_eye"])
+		possible_components += "Belligerent Eye"
+	if(clockwork_component_cache["vanguard_cogwheel"])
+		possible_components += "Vanguard Cogwheel"
+	if(clockwork_component_cache["guvax_capacitor"])
+		possible_components += "Guvax Capacitor"
+	if(clockwork_component_cache["replicant_alloy"])
+		possible_components += "Replicant Alloy"
+	if(clockwork_component_cache["hierophant_ansible"])
+		possible_components += "Hierophant Ansible"
+	if(!possible_components.len)
+		user << "<span class='warning'>[src] is empty!</span>"
+		return 0
+	var/component_to_withdraw = input(user, "Choose a component to withdraw.", name) as null|anything in possible_components
+	if(!user || !user.canUseTopic(src) || !component_to_withdraw)
```

Comment:
You need to also check if there's any of the chosen component left, though right now resource generation is bonkers so it doesn't matter too much.